core_contracts/stakedLp/stakedLp.py

```diff
@@ -0,0 +1,199 @@
+        self._governance.set(_dex)
+
+
+    @external(readonly=True)
+    def getAdmin(self) -> Address:
+        """
+        Gets the current admin address. This user can call using the
+        `@only_admin` decorator.
+        """
+        return self._governance.get()
+
+    @only_governance
+    @external
+    def setAdmin(self, _admin: Address) -> None:
+        """
+        :param _admin: The new admin address to set.
+        Can make calls with the `@only_admin` decorator.
+        Should be called before DEX use.
+        """
+        self._admin.set(_admin)
+
+    @only_admin
+    @external
+    def setRewards(self, _address: Address) -> None:
+        """
+        :param _address: New contract address to set.
+        Sets new Rewards contract address. Should be called before dex use.
+        """
+        if not _address.is_contract:
+            revert(f"{TAG}: Address provided is an EOA address. A contract address is required.")
+        self._rewards.set(_address)
+
+    @external(readonly=True)
+    def getRewards(self) -> Address:
+        """
+        Gets the address of the Rewards contract.
+        """
+        return self._rewards.get()
+
+    @external(readonly=True)
+    def balanceOf(self, _owner: Address, _id: int) -> int:
+        return self._poolStakeDetails[_owner][_id][Status.STAKED]
+
+    @external(readonly=True)
+    def totalSupply(self, _id: int) -> int:
+        return self._totalStaked[_id]
+
+    @only_governance
+    @external
+    def addPool(self, _id: int) -> None:
+        if not self._supportedPools[_id]:
+            self._supportedPools[_id] = True
+
+    @only_governance
+    @external
+    def removePool(self, _id: int) -> None:
+        if self._supportedPools[_id]:
+            self._supportedPools[_id] = False
+
+    def _stake(self, _user: Address, _id: int, _value: int) -> None:
+        # Validate inputs
+        pool_id = self._supportedPools[_id]
+        StakedLp._require(pool_id, f'pool with id:{_id} is not supported')
+        StakedLp._require(_value > 0, f'Cannot stake less than zero ,value to stake {_value}')
+
+        # Compute and store changes
+        previous_balance = self._poolStakeDetails[_user][_id][Status.STAKED]
+        previous_total = self._totalStaked[_id]
+        balance = previous_balance + _value
+        total = previous_total + _value
+        self._poolStakeDetails[_user][_id][Status.STAKED] = balance
+        self._totalStaked[_id] = total
+
+        # Define Interfaces
+        dex = self.create_interface_score(self._dex.get(), LiquidityPoolInterface)
+
+        # Get Source Name
+        name = dex.getPoolName(_id)
+
+        rewards = self.create_interface_score(self._rewards.get(), RewardInterface)
+        rewards.updateRewardsData(name, previous_total, _user, previous_balance)
+
+    @external
+    def unstake(self, _id: int, _value: int) -> None:
+        StakedLp._require(_id in self._supportedPools, f'pool with id:{_id} is not supported')
```

Comment:
Is this condition sufficient? Couldn't the key be in the DictDB, but the value be False? This is a more readable condition than the one on line 136, but the remove_pool method would need to be changed to delete the DictDB key.